tasks/store-flight/flightUi.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Entry point for the Flight task. Gathers parameters and performs validation.
+ */
+
+import inputHelper = require('../common/inputHelper');
+import request = require('../common/requestHelper');
+import fli = require('./flight');
+
+import path = require('path');
+
+import tl = require('vsts-task-lib');
+
+/** Obtain and validate parameters from task UI. */
+function gatherParams()
+{
+    var credentials: request.Credentials;
+    var endpointId = tl.getInput('serviceEndpoint', true);
+
+    /* Contrary to the other tl.get* functions, the boolean param here
+        indicates whether the parameter is __optional__ */
+    var endpointAuth = tl.getEndpointAuthorization(endpointId, false);
+    credentials =
+    {
+        tenant : endpointAuth.parameters['tenantId'],
+        clientId : endpointAuth.parameters['servicePrincipalId'],
+        clientSecret : endpointAuth.parameters['servicePrincipalKey']
+    };
+
+    var endpointUrl: string = endpointAuth.parameters['url'];
+    if (endpointUrl.lastIndexOf('/') == endpointUrl.length - 1)
+    {
+        endpointUrl = endpointUrl.substring(0, endpointUrl.length - 1);
+    }
+
+    var taskParams: fli.FlightParams = {
+        appName: '',
+        flightName: tl.getInput('flightName', true),
+        authentication: credentials,
+        endpoint: endpointUrl,
+        force: tl.getBoolInput('force', true),
+        zipFilePath: path.join(tl.getVariable('Agent.WorkFolder'), 'temp.zip'),
+        packages: []
+    };
+
+    // Packages
+    var packages: string[] = [];
+    if (inputHelper.inputFilePathSupplied('packagePath', false))
+    {
+        packages = packages.concat(inputHelper.resolvePathPattern(tl.getInput('packagePath', false)));
```

Comment:
For some reason in Typescript, push() does not support string[] as a param. It expects a string. It works in javascript but not in typescript. Compilation of the typescript fails if I use packages.push instead of packages.concat with this error: error TS2345: Argument of type 'string[]' is not assignable to parameter of type 'string'. Here is the simple example I created: https://www.typescriptlang.org/play/#src=var%20param%3A%20string%5B%5D%20%3D%20%5B%5D%3B%0D%0Aparam.push('yeah')%3B%0D%0Aalert(param)%3B